hooks/useFetchProjects.ts

```diff
@@ -94,7 +126,52 @@ const useFetchProjects = () => {
     fetchProjects();
   }, [projects, setProjects]);
 
-  return { projects, loading };
+  // handle tags extraction
+  useEffect(() => {
+    const newTags = extractTagsFromProjects(projects?.items?.items);
+    setTags(newTags);
+  }, [projects?.items?.items]);
+
+  // handle projects filtering by tags
+  useEffect(() => {
+    const projectsToDisplay = filterProjectsByTags(
+      projects?.items?.items,
+      tagToFilterBy
+    );
+    setFilteredProjects(projectsToDisplay);
+  }, [tags, tagToFilterBy, projects?.items?.items]);
+
+  // handle projects filtering by tags
+  useEffect(() => {
+    if (tags?.length) {
+      const newFiteredTags = structuredClone(tagToFilterBy);
```

Comment:
I would use a guard clause here just as you did before in the code to simplify it: 

```js
useEffect(() => {
  if (!tags?.length) return;
  rest of the code...
}
```

Reply:
return in useeffect acts a bit diffrently then expected in some use cases.
this is more readable and ensures no side effects.